openedx/core/djangoapps/theming/management/commands/create_sites_and_configurations.py

```diff
@@ -0,0 +1,180 @@
+    """
+    dns_name = None
+    theme_path = None
+    ecommerce_user = None
+    discovery_user = None
+
+    def add_arguments(self, parser):
+        """
+        Add arguments to the command parser.
+        """
+        parser.add_argument(
+            "--dns-name",
+            type=str,
+            help="Enter DNS name of sandbox.",
+            required=True
+        )
+
+        parser.add_argument(
+            "--theme-path",
+            type=str,
+            help="Enter theme directory path",
+            required=True
+        )
+
+    def _create_oauth2_client(self, url, site_name, is_discovery=True):
+        """
+        Creates the oauth2 client and add it in trusted clients.
+        """
+
+        client, _ = Client.objects.get_or_create(
+            redirect_uri="{url}complete/edx-oidc/".format(url=url),
+            defaults={
+                "user": self.discovery_user if is_discovery else self.ecommerce_user,
+                "name": "{site_name}_{client_type}_client".format(
+                    site_name=site_name,
+                    client_type="discovery" if is_discovery else "ecommerce",
+                ),
+                "url": url,
+                "client_id": "{client_type}-key-{site_name}".format(
+                    client_type="discovery" if is_discovery else "ecommerce",
+                    site_name=site_name
+                ),
+                "client_secret": "{client_type}-secret-{dns_name}".format(
+                    client_type="discovery" if is_discovery else "ecommerce",
+                    dns_name=self.dns_name
+                ),
+                "client_type": CONFIDENTIAL,
+                "logout_uri": "{url}logout/".format(url=url)
+            }
+        )
+        LOG.info("Adding {client} oauth2 client as trusted client".format(client=client.name))
+        TrustedClient.objects.get_or_create(client=client)
+
+    def _create_sites(self, site_domain, theme_dir_name, site_configuration):
+        """
+        Create Sites, SiteThemes and SiteConfigurations
+        """
+        site, created = Site.objects.get_or_create(
+            domain=site_domain,
+            defaults={"name": theme_dir_name}
+        )
+        if created:
+            LOG.info("Creating '{site_name}' SiteTheme".format(site_name=site_domain))
+            SiteTheme.objects.create(site=site, theme_dir_name=theme_dir_name)
+
+            LOG.info("Creating '{site_name}' SiteConfiguration".format(site_name=site_domain))
+            SiteConfiguration.objects.create(site=site, values=site_configuration, enabled=True)
+        else:
+            LOG.info(" '{site_domain}' site already exists".format(site_domain=site_domain))
+
+    def _get_sites_data(self):
+        """
+        Reads the json file from theme directory and returns the site data in JSON format.
+        """
+        site_data = {}
+        for site_name, site_dir in SITES.items():
+            configuration = json.dumps(
```

Comment:
per comment above, you should iterate over the dirs in edx-themes/edx-platform and load the sandbox_configuration.json if it exists, otherwise skip.